test/extensions/geoip_providers/maxmind/geoip_provider_test.cc

```diff
@@ -116,6 +115,7 @@ class GeoipProviderTestBase {
                                                           Filesystem::Watcher::OnChangedCb cb) {
                 {
                   absl::WriterMutexLock lock(&mutex_);
+                  on_changed_cbs_.reserve(1);
```

Comment:
Is the flake that we should only be accessing on_change_cbs_ with the mutex?

Reply:
yes, the tests were failing nondeterministically in somewhat random code paths before this fix (in release ci check)